### PR TITLE
move pkg/util/ipset inside pkg/proxy/ipvs

### PR DESCRIPTION
**Pull request by aroradaman**
#### What type of PR is this?
/kind cleanup

Follows code cleanup from https://github.com/kubernetes/kubernetes/pull/118296, https://github.com/kubernetes/kubernetes/pull/118296#issuecomment-1568714366

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/ipvs/ipset/ipset.go [hunk 1]
```diff
@@ -489,7 +491,7 @@ func validateProtocol(protocol string) bool {
 	if protocol == ProtocolTCP || protocol == ProtocolUDP || protocol == ProtocolSCTP {
 		return true
 	}
-	klog.Errorf("Invalid entry's protocol: %s, supported protocols are [%s, %s, %s]", protocol, ProtocolTCP, ProtocolUDP, ProtocolSCTP)
+	klog.ErrorS(validationError, "invalid protocol", "protocol", protocol, "supported_protocols", fmt.Sprintf("[%s, %s, %s]", ProtocolTCP, ProtocolUDP, ProtocolSCTP))
```
Comment (uablrek) [inline pkg/proxy/ipvs/ipset/ipset.go:494]: Don't use "Sprintf", use a structured variable, and drop the underscore:
```go
	klog.ErrorS(validationError, "invalid protocol", "protocol", protocol, "supported protocols", []string{ProtocolTCP, ProtocolUDP, ProtocolSCTP})

```

**Review by uablrek — APPROVED** [review]

File: pkg/proxy/ipvs/ipset/ipset.go [hunk 2]
```diff
@@ -198,22 +200,22 @@ func (e *Entry) Validate(set *IPSet) bool {
...
 		}
 		begin, end, err := parsePortRange(set.PortRange)
 		if err != nil {
-			klog.Errorf("Failed to parse set %v port range %s for ipset %v, error: %v", set, set.PortRange, set, err)
+			klog.ErrorS(err, "failed to parse set port range", "ipset", set, "port_range", set.PortRange)
 			return false
 		}
 		if e.Port < begin || e.Port > end {
-			klog.Errorf("Entry %v port number %d is not in the port range %s of its ipset %v", e, e.Port, set.PortRange, set)
+			klog.ErrorS(validationError, "port number is not in the port range of its ipset", "entry", e, "port", e.Port, "port_range", set.PortRange, "ipset", set)
```
[the remark this answers is not in the document]
Comment (danwinship) [reply pkg/proxy/ipvs/ipset/ipset.go]: I don't know the conventions for structured logging keys...

[the remark this answers is not in the document]
Comment (aroradaman) [reply pkg/proxy/ipvs/ipset/ipset.go]: I have updated structured logging keys with camel casing, **pkg/proxy/ipvs** uses the same convention and is quite common among other modules.

**Review by uablrek — APPROVED** [review]

**Review by uablrek — APPROVED** [review]

**Comment by uablrek** [comment]
@thockin pkg/util/ipset needs lgtm from sig-network-approvers (moved to proxy/ipvs by this PR)

**Pull request merged**